test/e2e-secondary-network/infra/prepare_cluster_common.sh

```diff
@@ -0,0 +1,115 @@
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+function generateAntreaConfig() {
+        genManifest=$THIS_DIR/../../../../hack/generate-manifest.sh
+	if [ "$1" == "sriov" ]; then
+        	if [ "$sriov_interface" != '' ] && [ "$vfs_num" != '' ]; then
+                	$genManifest --sriov --whereabouts > $YAML_ANTREA
+		fi
+	
+	elif [ "$1" == "veth" ]; then
+                $genManifest --whereabouts > $YAML_ANTREA
+        fi
+
+        if [ $? -ne 0 ]; then
+              echoerr "Failed to generate Antrea.yml"
+              exit 1
+        fi
+        echo "Antrea.yml generated with required secondary network configuration parameters."
+	sed -i 's/#  \([[:space:]]*SecondaryNetwork.*\)/\1/' $YAML_ANTREA 
+	if [ $? -ne 0 ]; then
+              echoerr "Failed to update Antrea"
+              exit 1
+        fi
+	sleep 2
+	sed -i '/^\([[:space:]]*#SecondaryNetwork: *\).*/s## true#g' $YAML_ANTREA
+}
+#ARUN - common
+function updateKubeConfigForTargetCluster() {
+        mkdir -p /tmp/kubernetes/$hostIP
+        scp $hostIP:/etc/kubernetes/admin.conf /tmp/kubernetes/$hostIP/.
+        if [ $? -ne 0 ]; then
+                echoerr "Failed to copy the downloaded configuration files!"
+                exit 1
+        fi
+        echo export KUBECONFIG="/tmp/kubernetes/$hostIP/admin.conf" > env.sh
+        source ./env.sh
+        #echo "kubeconfig updated with the remote cluster configuration!"
+}
+
+function AntreaCNI() {
+        kubectl $action -f $YAML_ANTREA
+        if [ $? -ne 0 ]; then
+              echoerr "Failed to $action Antrea CNI"
+              exit 1
+        fi
+        IFS=$'\n' read -r -d '' -a antrea_agents < <(kubectl get pods -n $NAMESPACE --no-headers -o custom-columns=":metadata.name" | grep antrea-agent && printf '\0')
+        for agent in ${antrea_agents[@]};do
+                iterate=0
+                while [ $iterate -ne 30 ]; do
+                        if [[ $(kubectl get pods -n $NAMESPACE $agent -o 'jsonpath={..status.conditions[?(@.type=="Ready")].status}') != "True" ]]
+                        then
+                                echoerr "waiting for $agent to be ready."
+                                sleep 1
+                        else
+                                echoerr "$agent is ready"
+                                break
+                        fi
+                        iterate=$((iterate+1))
+                done
+        done
+        if [[ $iterate -eq 30 ]]; then
+              echoerr "$agent Antrea CNI deployment failed or timedout"
+              exit 1
+        fi
```

Comment:
can't you use `kubectl -n kube-system rollout status ds/antrea-agent` instead of all this?

Reply:
yes. got it aligned.